### ISSUE TITLE
False positive NoValue 

### ISSUE THREAD

**Issue by mrsuh** [vimeo/psalm#9349]
Hi.
Psalm 5.7.1
I got this error on a simple example.
```php
<?php

$str = $argv[1] ?? '';
if(empty($str) || strlen($str) < 3) {
    exit(1);
}

echo $str; // ERROR: [NoValue](https://psalm.dev/179) - 8:6 - All possible types for this argument were invalidated - This may be dead code
```
https://psalm.dev/r/f12fbc1db8

The error disappeared when I removed the unnecessary "empty" check.
```php
<?php

$str = $argv[1] ?? '';
if(strlen($str) < 3) {
    exit(1);
}

echo $str;
```
https://psalm.dev/r/a0bf807db5

**Comment by piotrekkr** [comment]
Same with:
```php
<?php
class CodeReader
{
    public function read(string $code): ?string
    {
        if (empty($code) || strlen($code) < 32) {
            return null;
        }

        $code = substr($code, 3);
        
        return $code;
    }
}
```
 https://psalm.dev/r/a99bcc7c07

works without empty
```php
<?php
class CodeReader
{
    public function read(string $code): ?string
    {
        if (strlen($code) < 32) {
            return null;
        }

        $code = substr($code, 3);
        
        return $code;
    }
}
```
https://psalm.dev/r/1fbf4fbcfc

**Comment by jacekkarczmarczyk** [comment]
slighly different usecase with different error, but I guess underlying issue is the same

https://psalm.dev/r/01a80b3670

**Comment by jack-worman** [comment]
Here is another case that I hit: https://psalm.dev/r/b72669f3a0

**Comment by weirdan** [comment]
Thanks @jack-worman, this is probably the clearest example out of all posted here.

With traces: https://psalm.dev/r/8b1462cca9
